syntax = "proto3";

option java_multiple_files = false;
option java_package = "grpc.document";
option java_outer_classname = "Generation";

package grpc.document;

// The Generation service definition.
service TopicGeneration {
  // Sends a request to get topics from a file
  rpc GenerateTopicsDirect (FileLocation) returns (Topics) {}

  rpc GenerateTopicsIndirect (FileLocation) returns (FileLocation) {}
}

// The request message containing the user's name.
message FileLocation {
  string location = 1;
}

// The response message containing the greetings
message Topics {
  repeated string topics = 1;
}
